docs/FAQ.md

```diff
@@ -0,0 +1,236 @@
+- [Apollo Server](https://github.com/apollographql/apollo-server/) explicitly checks that integers are at most 32 bits
+- [gqlgen](https://github.com/99designs/gqlgen) by default allows any integer that fits in `int` (i.e. 64 bits on most platforms)
+- [Apollo Client](https://github.com/apollographql/apollo-client) doesn't check (but implicitly is limited to 53 bits by JavaScript)
+- [shurcooL/graphql](https://github.com/shurcooL/graphql) requires integers be passed as a `graphql.Int`, defined to be an `int32`
+
+By default, genqlient maps GraphQL `Int`s to Go's `int`, meaning that on 64 bit systems there's no client-side restriction.  If you prefer to limit integers to `int32`, you can set a binding in your `genqlient.yaml`:
+
+```yaml
+bindings:
+  Int:
+    type: int32
+```
+
+Or, you can bind it to any other type, perhaps one with size-checked constructors; see the [`genqlient.yaml` documentation](`genqlient.yaml`) for more details.
+
+## How do I make a query with …
+
+### … a specific name for a field?
+
+genqlient supports GraphQL field-aliases, and uses them to determine the Go struct field name.  For example, if you do
+```graphql
+query MyQuery {
+  myGreatName: myField
+}
+```
+and genqlient will generate a Go field `MyGreatName string`.  Note that the alias will always be uppercased, to ensure the field is visible to the Go JSON library.
+
+### … nullable fields?
+
+There are two ways to handle nullable fields in genqlient.  One way is to use the Go idiom, where null gets mapped to the zero value; this is the default in genqlient.  So if you have a GraphQL field of type `String`, and you do:
+
+```graphql
+query MyQuery(arg: String) {
+  myField
+}
+```
+
+then genqlient will generate a Go field `MyField string`, and set it to the empty string if the server returns null.  This works even for structs: if an object type in GraphQL is null, genqlient will set the corresponding struct to its zero value.  It can be helpful to request `id` in such cases, since that’s a field that should always be set, or `__typename` which is guaranteed to be set, so you can use its presence to decide whether to look at the other fields.
+
+For input fields, you often want to tell genqlient to send null to the server if the argument is set to the zero value, similar to the JSON `omitempty` tag.  In this case, you can do:
+
+```graphql
+query MyQuery(
+  # @genqlient(omitempty: true)
+  arg: String,
+) {
+  myField
+}
+```
+
+You can also put the `# @genqlient(omitempty: true)` on the first line, which will apply it to all arguments in the query.
+
+If you need to distinguish null from the empty string (or generally from the Go zero value of your type), you can tell genqlient to use a pointer for the field or argument like this:
+```graphql
+query MyQuery(
+  # @genqlient(pointer: true)
+  arg: String,
+) {
+  # @genqlient(pointer: true)
```

Comment:
Can we also put `# @genqlient(pointer: true)` on the first line to apply it to everything in the query, similar to the previous omitempty example?

Reply:
Yes!  Will mention.